tests/src/Unit/DataService/NotificationListenerTest.php

```diff
@@ -0,0 +1,104 @@
+use Lullabot\Mpx\Service\IdentityManagement\UserSession;
+use Lullabot\Mpx\Tests\JsonResponse;
+use Lullabot\Mpx\Tests\MockClientTrait;
+use Lullabot\Mpx\TokenCachePool;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * Tests listening for notifications.
+ *
+ * @coversDefaultClass  \Lullabot\Mpx\DataService\NotificationListener
+ */
+class NotificationListenerTest extends TestCase
+{
+    use MockClientTrait;
+
+    /**
+     * Test fetching the last sync ID.
+     *
+     * @covers ::__construct
+     * @covers ::sync
+     */
+    public function testSync()
+    {
+        $id = rand(1, getrandmax());
+        $client = $this->getMockClient([
+            new JsonResponse(200, [], 'signin-success.json'),
+            new JsonResponse(200, [], 'resolveAllUrls.json'),
+            new JsonResponse(200, [], [
+                [
+                    'id' => $id,
+                ],
+            ]),
+        ]);
+        $user = new User('username', 'password');
+        $tokenCachePool = new TokenCachePool(new ArrayCachePool());
+        /** @var StoreInterface|\PHPUnit_Framework_MockObject_MockObject $store */
+        $store = $this->getMockBuilder(StoreInterface::class)->getMock();
+        $session = new UserSession($user, $client, $store, $tokenCachePool);
+        $authenticatedClient = new AuthenticatedClient($client, $session);
+        $manager = DataServiceManager::basicDiscovery();
+        $service = $manager->getDataService('Media Data Service', 'Media', '1.10');
+        $listener = new NotificationListener($authenticatedClient, $service, 'unit-tests');
+        $last = $listener->sync()->wait();
+        $this->assertEquals($id, $last);
+    }
+
+    /**
+     * Test listening for notifications.
+     *
+     * @covers ::__construct
+     * @covers ::listen
+     */
+    public function testListen()
+    {
+        $client = $this->getMockClient([
+            new JsonResponse(200, [], 'signin-success.json'),
+            new JsonResponse(200, [], 'resolveAllUrls.json'),
+            new JsonResponse(200, [], 'notification.json'),
+        ]);
+        $user = new User('username', 'password');
+        $tokenCachePool = new TokenCachePool(new ArrayCachePool());
+        /** @var StoreInterface|\PHPUnit_Framework_MockObject_MockObject $store */
+        $store = $this->getMockBuilder(StoreInterface::class)->getMock();
+        $session = new UserSession($user, $client, $store, $tokenCachePool);
+        $authenticatedClient = new AuthenticatedClient($client, $session);
+        $manager = DataServiceManager::basicDiscovery();
+        $service = $manager->getDataService('Media Data Service', 'Media', '1.10');
+        $listener = new NotificationListener($authenticatedClient, $service, 'unit-tests');
+        /** @var \Lullabot\Mpx\DataService\Notification[] $notifications */
+        $notifications = $listener->listen(98765)->wait();
+        $this->assertCount(2, $notifications);
```

Comment:
Why this should be 2?

Reply:
Unfortunately, the structure of notifications is slightly different from object lists. Where object lists have something like:

```json
{
  entryCount: 2,
  entries: [
    { // an array of objects, such as media items, all in one "list" },
  ]
}
```

Notifications instead have an outer array only, with objects inside. See [notification.json](https://github.com/Lullabot/mpx-php/blob/e20144c91e705c87a8a88ec4d506cfe27b938c72/tests/fixtures/notification.json).

So in this case, we test MPX returning multiple notifications in one response (it can return up to the amount specified in the notification call).

Reply:
I didn't see the amount of 2 in the notification call. You mean listen() right? I don't see a second parameter there.

Reply:
Yes, "listen" eventually returns an array of Notifications. The individual HTTP response that's mocked contains two notifications (in production, you'd generally have anywhere from 1 to 500 items, or a timeout would trigger).